OutOfSchool/OutOfSchool.WebApi.Tests/Services/DraftStorage/DraftStorageServiceTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models.Workshops;
+using OutOfSchool.BusinessLogic.Services.DraftStorage;
+using OutOfSchool.BusinessLogic.Services.DraftStorage.Interfaces;
+using OutOfSchool.Redis;
+
+namespace OutOfSchool.WebApi.Tests.Services.DraftStorage;
+
+[TestFixture]
+public class DraftStorageServiceTests
+{
+    private Mock<IReadWriteCacheService> readWriteCacheServiceMock;
+    private Mock<ILogger<DraftStorageService<WorkshopBaseDto>>> loggerMock;
+    private IDraftStorageService<WorkshopBaseDto> draftStorageService;
+
+    [SetUp]
+    public void SetUp()
+    {
+        loggerMock = new Mock<ILogger<DraftStorageService<WorkshopBaseDto>>>();
+        readWriteCacheServiceMock = new Mock<IReadWriteCacheService>();
+        draftStorageService = new DraftStorageService<WorkshopBaseDto>(readWriteCacheServiceMock.Object, loggerMock.Object);
+    }
+
+    [Test]
+    public async Task RestoreAsync_WhenDraftExistsInCache_ShouldRestoreAppropriatedEntity()
+    {
+        // Arrange
+        var workshopDraft = new WorkshopBaseDto()
+        {
+            Title = "title",
+            Email = "myemail@gmail.com",
+            Phone = "+380670000000",
+        };
+        var expected = new Dictionary<string, string>()
+            {
+                {"ExpectedKey", "{\"Title\":\"title\",\"Email\":\"myemail@gmail.com\",\"Phone\":\"+380670000000\"}"},
```

Comment:
Please cosider refactoring (`JsonConvert.SerializeObject` and usage of `const`)

There are two issues:
1. Manual json serialization. Wish a good luck for next developer trying to change `workshopDraft`
2. Magic string - `"ExpectedKey"` - used here and in `Setup` and in `RestoreAsync`. Another "good luck wishes" for changing this test.

Reply:
Ok, done.
Used Bogus random string for generating key and cacheKey instead hardcore string.